subprojects/platform-native/src/docs/guide-templates/building-native-application/README.adoc

```diff
@@ -0,0 +1,172 @@
+├── gradlew      // <2>
+└── gradlew.bat  // <2>
+
+2 directories, 4 files
+$ touch settings.gradle // <3>
+
+----
+
+<1> Generated folder for wrapper files
+
+<2> Gradle wrapper start scripts
+
+<3> Create an empty Gradle workspace buildscript
+
+We now have a Gradle project setup.
+Let's configure {C application} capability to the project.
+
+== 2. Configure {C application} capability
+
+First, create the project buildscript, e.g. `build.gradle`.
+
+build.gradle
+
+[source,groovy]
+----
+plugins {  // <1>
+	id 'dev.nokee.c-application' // <2>
+}
+----
+
+<1> We use the plugin block to apply plugins that extends the project capability.
+We can apply any Gradle project plugin here.
+Learn more about the link:TODO[plugins block in Gradle's documentation].
+Refer to each plugin's reference chapter to learn more about each individual plugins.
+
+<2> We apply Nokee's {C application} plugin.
+Learn more about link:TODO[this plugin at its reference chapter].
+
+NOTE: Gradle support two domain specific language (DSL): Groovy DSL and Kotlin DSL.
+For this guide, we will use Groovy DSL which is closer in syntax to {C} language.
+Learn more about Gradle DSL language in the link:TODO[Getting started chapter].
+
+== 3. Add {C} code
+
+By default, Nokee uses link:TODO[Gradle's conventional source layout].
+All {C} application code is located at `src/main/c` and its private headers are located at `src/main/headers`.
+
+Let's add a very simple hello world:
+
+.src/main/c/main.c
+[source,c]
+----
+#include <stdio>
+#include <my_string.h>
+
+int main(int argc, char** argv) {
+	printf("%s\n", kHelloWorld);
+	return 0;
+}
+----
+
+.src/main/headers/my_string.h
+[source,c]
+----
+static const char* kHelloWorld = "Hello, World!";
+----
+
+Next, let's build the application.
+
+== 3. Build the {application}
+
+[source,terminal]
+----
+$ ./gradlew build
+
+BUILD SUCCESSFUL
+7 actionable tasks: 7 executed
+----
+
+NOTE: The first time you run the wrapper script, e.g. `gradlew[.bat]`, there may be a delay while that version of gradle is downloaded and stored locally in your `~/.gradle/wrapper/dists` folder.
+Learn link:TODO[why Gradle download distribution when using wrapper].
+
+The build task compiles the {C} sources, links the object files, and runs the tests, if any.
+The following shows the content of the `build` folder:
+
+[source,terminal]
+----
+$ tree ./build
+./build
+|- exe
+  |- main
+    |- app // <1>
+|- objs  // <2>
+  |- main
+    |- <hash>
+      |- main.o
+|- tmp // <3>
+  |- compileC
+    |- options.txt
+  |- link
+    |- options.txt
+----
+<1> The built executable
+<2> All object files are built here
+<3> The `tmp` folder is a temporary workspace for all Gradle tasks.
+You can find `options.txt` file containing all flags for both compile and link tasks.
+
+NOTE: Dependencies on other projects isn’t covered in this guide.
+To learn more about this subject, have a look at the transitive dependency sample for a demonstration.
```

Comment:
🔲 We should call out samples and another guide that explains how to work with dependencies.